lmfit/minimizer.py

```diff
@@ -1427,7 +1429,19 @@ def least_squares(self, params=None, **kws):
 
             # calculate the cov_x and estimate uncertainties/correlations
             try:
-                hess = np.matmul(ret.jac.T, ret.jac)
+                if issparse(ret.jac):
+                    hess = (ret.jac.T * ret.jac).toarray()
+                elif isinstance(ret.jac, LinearOperator):
+                    I = np.eye(ret.jac.shape[1], dtype=ret.jac.dtype)
+                    # Can remove check for scipy >= 1.4.0
```

Comment:
please clarify this comment; which check can be removed when using scipy >= 1.4.0? Can one remove the ```try....except``` block and just always do ```hess = (ret.jac.T * ret.jac) * I```, or ...? Also please add a TODO or FIXME marker so that it shows up easily with things like ```pylint``` and others.